eessi/testsuite/tests/apps/osu.py

```diff
@@ -39,25 +41,23 @@ def filter_scales_coll():
     ]
 
 
-@rfm.simple_test
-class EESSI_OSU_Micro_Benchmarks_pt2pt(osu_benchmark):
-    ''' Run-only OSU test '''
-    scale = parameter(filter_scales_pt2pt())
-    valid_prog_environs = ['default']
-    valid_systems = ['*']
+class EESSI_OSU_Base(osu_benchmark):
+    """ base class for OSU tests """
     time_limit = '30m'
     module_name = parameter(find_modules('OSU-Micro-Benchmarks'))
-    # Device type for non-cuda OSU-Micro-Benchmarks should run on hosts of both node types. To do this the default
-    # device type is set to GPU.
     device_type = parameter([DEVICE_TYPES[CPU], DEVICE_TYPES[GPU]])
-    # unset num_tasks_per_node from the hpctestlib.
+
+    # reset num_tasks_per_node from the hpctestlib: we handle it ourselves
     num_tasks_per_node = None
 
     # Set num_warmup_iters to 5 to reduce execution time, especially on slower interconnects
     num_warmup_iters = 5
     # Set num_iters to 10 to reduce execution time, especially on slower interconnects
     num_iters = 10
 
+    def required_mem_per_node(self):
+        return 1024
```

Comment:
Wait, this is not enough, is it? I seem to remember these tests using a surprising amount of memory in the past, I guess because of the buffers for the collective tests. Not entirely sure though, I don't fully remember... Did you actually measure the memory usage for this one, for various task counts?

Reply:
Comment from Sam: he did measure memory, and it was less than 1GB for all tests.
Comment from Satish: buffers used to be retained between iterations in older versions, in newer versions this should be fixed.

Reply:
it's weird that for me the used memory is always ~ the same value independent of the number of cores. also no difference between srun and mpirun.

i changed it to 1GB per task, which should be more than enough according to an allreduce test of Caspar in Snellius.